### DISCUSSION TITLE
How to enable additional Function runtimes?

### DISCUSSION THREAD

**Discussion by tanmoy27112000** [appwrite/appwrite#4826] [Q&A] [answered]
### 👟 Reproduction steps

I am using M1 MacBook Air and when I create a function it does not show the dart runtime.
install the appwrite CLI and use appwrite init function
![Screenshot 2022-12-08 at 2 28 54 AM](https://user-images.githubusercontent.com/22212661/206294600-a4eb490e-1144-4ca9-b208-4affecac9e5e.png)

### 👍 Expected behavior

It should show the dart runtime 

### 👎 Actual Behavior

It does not show dart runtime

### 🎲 Appwrite version

Version 1.1.x

### 💻 Operating system

MacOS

### 🧱 Your Environment

localhost

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment, marked as answer]
By default, those are the only runtimes enabled. To enable additional runtimes, you can update the `_APP_FUNCTIONS_RUNTIMES` environment variable (keep in mind the `.env` file is hidden on Unix systems). 

For a list of supported runtimes, refer to the [Functions docs](https://appwrite.io/docs/functions#supportedRuntimes). 

For information on environment variables, refer to the [Environment Variables docs](https://appwrite.io/docs/environment-variables).

**Reply by sgruhier** [reply]
I have this 
```
_APP_FUNCTIONS_RUNTIMES=php-8.0,node-16.0,python-3.9,ruby-3.0,dart-2.17
``` 

But I still don't see dart in my runtimes. 

I use digitalocean image. I also added dart runtime in docker like this 
```
docker pull openruntimes/dart:v2-2.17
``` 

Which step I am missing?
thanks
Seb

**Reply by sgruhier** [reply]
I forgot to run 
```
docker compose up -d
```
RTFM XD :)

**Reply by fredrikbaberg** [reply]
Just wanted to note that this differs from the [documentation](https://appwrite.io/docs/functions#supportedRuntimes) that states: 
> By default, the following runtimes are enabled: node-16.0, php-8.0, python-3.9, ruby-3.0, and dart-2.17.

With that said, the steps above works:
* Update `_APP_FUNCTIONS_RUNTIMES` with desired runtimes.
* Run `docker-compose up -d`.

I also did `docker pull openruntimes/dart:v2-2.17`, not sure that is necessary.
